test/image_test.py

```diff
@@ -750,9 +750,52 @@ def f(self):
         print("bar!", VARIABLE_6)
 
 
+class FooInstance:
+    not_used_by_build_method: str = "normal"
+    used_by_build_method: str = "normal"
+
+    @build()
+    def build_func(self):
+        global VARIABLE_5
+
+        print("global variable", VARIABLE_5)
+        print("static class var", FooInstance.used_by_build_method)
+
+
+def test_image_cls_var_rebuild(client, servicer):
+    rebuild_app = App()
+    image_id = -1
+    rebuild_app.cls(image=Image.debian_slim())(FooInstance)
+    with rebuild_app.run(client=client):
+        image_id = list(servicer.images)[-1]
+    FooInstance.used_by_build_method = "rebuild"
+    rebuild_app.cls(image=Image.debian_slim())(FooInstance)
+    with rebuild_app.run(client=client):
+        image_id2 = list(servicer.images)[-1]
+    FooInstance.used_by_build_method = "normal"
```

Comment:
Did you intend to check that this doesn't rebuild _again_ after you revert the change? That seems like a good idea if so.

Reply:
Oh sure good point